src/shared/acpi-fpdt.c

```diff
@@ -61,6 +61,36 @@ struct acpi_fpdt_boot {
         uint64_t exit_services_exit;
 } _packed;
 
+/* /dev/mem is deprecated on many systems try using /sys/firmware/acpi/fpdt parsing instead */
+static int acpi_get_boot_usec_kernel_parsed(usec_t *ret_loader_start, usec_t *ret_loader_exit) {
+        usec_t nsecs;
+        int r;
+
+        r = read_timestamp_file("/sys/firmware/acpi/fpdt/boot/exitbootservice_end_ns", &nsecs);
+        if (r < 0)
+                return r;
+
+        if (nsecs == 0)
+                /* Non-UEFI compatible boot. */
+                return -ENODATA;
+
+        r = read_timestamp_file("/sys/firmware/acpi/fpdt/boot/bootloader_load_ns", &nsecs);
+        if (r < 0)
+                return r;
+
+        if (ret_loader_start)
+                *ret_loader_start = nsecs / 1000;
```

Comment:
according to our coding style return parameters should either all be set on success, or not touched at all on failure. but here you set this return parameter, but then there's a codepath below where we still can fail.

see: https://systemd.io/CODING_STYLE/

Reply:
Yah, I'm going to have to store both values to validate exit>start as well. So that will fix this problem too.

Reply:
This logic follows the /dev/mem version almost exactly now, so this should be cleared up as well.